src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,128 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] objects = new Object[DEFAULT_CAPACITY];
+    private int size = 0;
+
+    private Object[] grow() {
+        return grow(size + 1);
+    }
+
+    private Object[] grow(int minCapacity) {
+        return objects = Arrays.copyOf(objects, newCapacity(minCapacity));
+    }
+
+    private int newCapacity(int minCapacity) {
+        int oldCapacity = objects.length;
+        return (int) (oldCapacity + (oldCapacity * 1.5));
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < objects.length) {
+            objects[size] = value;
+            size++;
+        } else {
+            objects = grow();
+            if (size < objects.length) {
+                objects[size] = value;
+                size++;
+            }
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("The index is bigger than"
+                   + " array size or less than null");
+        }
+        if (size == objects.length) {
+            objects = grow();
+        }
+        System.arraycopy(objects, index,
+                objects, index + 1,
+                size - index);
+        objects[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] inputObjects = new Object[list.size()];
+        for (int i = 0; i < list.size(); i++) {
+            inputObjects[i] = list.get(i);
+        }
+        if ((objects.length - size) < inputObjects.length) {
+            objects = grow();
+        }
+        System.arraycopy(inputObjects, 0, objects, size, inputObjects.length);
+        size += inputObjects.length;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounce exception");
+        }
```

Comment:
1. Create separate private method for this check:
```suggestion
        checkIndex(int index);
```
2. Make exception message more informative : put arguments there:
"throw new ArrayListIndexOutOfBoundsException("Index:" + index + "out of size" + size)"

Reply:
done